connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -355,24 +357,44 @@ private WorkerTask buildWorkerTask(ConnectorConfig connConfig,
                                        Converter valueConverter) {
         // Decide which type of worker task we need based on the type of task.
         if (task instanceof SourceTask) {
+            // merging connector config with worker config to apply producer overrides
+            // on the connector
+            WorkerConfig mergedConfig = getMergedConfig(config, connConfig, "producer");
             TransformationChain<SourceRecord> transformationChain = new TransformationChain<>(connConfig.<SourceRecord>transformations());
             OffsetStorageReader offsetReader = new OffsetStorageReaderImpl(offsetBackingStore, id.connector(),
                     internalKeyConverter, internalValueConverter);
             OffsetStorageWriter offsetWriter = new OffsetStorageWriter(offsetBackingStore, id.connector(),
                     internalKeyConverter, internalValueConverter);
             KafkaProducer<byte[], byte[]> producer = new KafkaProducer<>(producerProps);
```

Comment:
Maybe I'm missing something, but don't the overrides need to somehow make it to `producerProps`?